langchain/vectorstores/mongodb_atlas.py

```diff
@@ -0,0 +1,242 @@
+            docs.append((Document(page_content=text, metadata=res), score))
+        return docs
+
+    def similarity_search(
+        self,
+        query: str,
+        k: int = 4,
+        pre_filter: Optional[dict] = None,
+        post_filter_pipeline: Optional[List[Dict]] = None,
+        **kwargs: Any,
+    ) -> List[Document]:
+        """Return MongoDB documents most similar to query.
+
+        Use the knnBeta Operator available in MongoDB Atlas Search
+        This feature is in early access and available only for evaluation purposes, to
+        validate functionality, and to gather feedback from a small closed group of
+        early access users. It is not recommended for production deployments as we may
+        introduce breaking changes.
+        For more: https://www.mongodb.com/docs/atlas/atlas-search/knn-beta
+
+        Args:
+            query: Text to look up documents similar to.
+            k: Optional Number of Documents to return. Defaults to 4.
+            pre_filter: Optional Dictionary of argument(s) to prefilter on document
+                fields.
+            post_filter_pipeline: Optional Pipeline of MongoDB aggregation stages
+                following the knnBeta search.
+
+        Returns:
+            List of Documents most similar to the query and score for each
+        """
+        docs_and_scores = self.similarity_search_with_score(
+            query,
+            k=k,
+            pre_filter=pre_filter,
+            post_filter_pipeline=post_filter_pipeline,
+        )
+        return [doc for doc, _ in docs_and_scores]
+
+    @classmethod
+    def from_texts(
+        cls,
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        connection_string: Optional[str] = None,
+        namespace: Optional[str] = None,
+        **kwargs: Any,
+    ) -> MongoDBAtlasVectorSearch:
+        """Construct MongoDBAtlasVectorSearch wrapper from raw documents.
+
+        This is a user-friendly interface that:
+            1. Embeds documents.
+            2. Adds the documents to a provided MongoDB Atlas Vector Search index
+                (Lucene)
+
+        This is intended to be a quick way to get started.
+
+        Example:
+            .. code-block:: python
+
+                from langchain.vectorstores import MongoDBAtlasVectorSearch
+                from langchain.embeddings import OpenAIEmbeddings
+
+                connection_string = "<YOUR-CONNECTION-STRING>"
+                namespace = "<db_name>.<collection_name>"
+                embeddings = OpenAIEmbeddings()
+                vectorstore = MongoDBAtlasVectorSearch.from_texts(
+                    texts,
+                    embeddings,
+                    meadatas=metadatas,
+                    connection_string=connection_string,
+                    namespace=namespace
+                )
+        """
+        if not connection_string or not namespace:
+            raise ValueError(
```

Comment:
I find it weird to define default None values for mandatory parameters and then making sure here that they have to be passed as non None values.
This is why I used kwargs in my initial commit since the vectorstore abstractmethod is a blocking element. I would recommend to make this abstractmethod more flexible to avoid such situation.

Reply:
changing abstract method at this point would be a large (potentially breaking) change, so would want to be very thoughtful about that and would warrant it's own pr. what would the ideal interface look like in your opinion?

Reply:
`metadatas: Optional[List[dict]] = None` could be  `metadatas: Optional[List[dict]]`which would allow for other following parameters without default values